src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,103 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Class which parses commands and executes code based off the commands.
+ *
+ */
+public class Parser {
+
+    /** Returns String date in a nicer format.
+     *
+     * @param date
+     * @return String date in nicer format
+     */
+    public String getDate(String date) {
+        LocalDate localDate = LocalDate.parse(date);
+        return localDate.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
+    }
+
+    /**
+     * Parses the command and executes code based off the command.
+     *
+     * @param fullCommand
+     * @param ui
+     * @param taskList
+     * @param storage
+     * @return true if "bye" command was given
+     */
+    public boolean parse(String fullCommand, Ui ui, TaskList taskList, Storage storage) {
+        String[] commandBreakdown = fullCommand.split(" ");
+        String command = commandBreakdown[0];
+
+        switch (command) {
+        case "list":
```

Comment:
It's great that you follow the coding standard for case clauses in switch statement.